### PR TITLE
AO3-3867 AO3-6194 Improve managing rejected and unreviewed collection items

### PR DESCRIPTION
**Pull request by sarken**
## Issue

https://otwarchive.atlassian.net/browse/AO3-3867
https://otwarchive.atlassian.net/browse/AO3-6194 

## Purpose

Updates #3837

This updates the manage collection item pages rather than a user's works in collections page because doing the latter would require us to either stop using Elasticsearch for those pages or add more things to Elasticsearch.

This also allows alters collection moderators can see on the collection's manage items page because... I have no idea if we decided it was necessary for some reason or if it's just a case of "they both use this page, so it's easier to show them both things" (spoiler: it wasn't easier, because then I had to change the permissions for removing items).

* Lets users can see collection items awaiting collection approval or rejected by collections.
* Lets collection moderators can see items rejected by the user.
* Prevents collection moderators from removing items rejected by the item's creator (because they might re-add the item to annoy the creator), unless they are also the item's creator.
* Includes select menus for both collection and user approval status on the collection items pages, disabling the one the current user cannot change (e.g. if a user is on their collection items page, the select menu showing the collection approval status is visible but disabled because only the collection moderator can change this). Also adds invisible labels for these select menus.
* Puts the checkbox in front of the text on the "Revealed" and "Anonymous" checkboxes.
* Adds styling fixes for the selects and checkboxes for the default skin and Reversi.
* Makes the text of the collection items index translatable.
* Removes incorrect ARIA role on navigation.
* Exposes pagination on user_collection_items page. (AO3-6194)

It does not [update the links in the emails](https://github.com/otwcode/otwarchive/pull/3837#pullrequestreview-416544138) beyond pointing to the new URL for the equivalent pages. We can add `works/:id/collection_items` pages in a separate issue and update the emails then.

## Testing Instructions

Refer to Jira.

## References

#3837

## Credit

Sarken, she/her

### CODE REVIEW

File: app/controllers/collection_items_controller.rb [hunk 1]
```diff
@@ -9,23 +9,29 @@ def index
...
+                          when "rejected_by_collection"
                             @collection_items.rejected_by_collection
-                          when params[:invited]
+                          when "rejected_by_user"
+                            @collection_items.rejected_by_user
+                          when "unreviewed_by_user"
                             @collection_items.invited_by_collection
                           else
                             @collection_items.unreviewed_by_collection
                           end
```
[the remark this answers is not in the document]
Comment (sarken) [reply app/controllers/collection_items_controller.rb:23]: We can make an issue and I'll add a TODO!

File: app/models/collection_item.rb [hunk 2]
```diff
@@ -165,21 +165,34 @@ def item_date
...
+      (self.collection.user_is_maintainer?(user) && !self.rejected_by_user?)
   end
 
   def approve_by_user
     self.user_approval_status = :approved
   end
 
+  def reject_by_user
+    self.user_approval_status = :rejected
+  end
```
[the remark this answers is not in the document]
Comment (sarken) [reply app/models/collection_item.rb:178]: Oh, they're not -- I was using them from the console. I will take them out!

File: features/step_definitions/collection_steps.rb [hunk 3]
```diff
@@ -42,14 +42,18 @@
...
-    visit collection_items_path(c, invited: true)
+    visit collection_items_path(c, status: "approved")
+  elsif item_status == "rejected by user"
+    visit collection_items_path(c, status: "rejected_by_user")
+  elsif item_status == "rejected by collection"
+    visit collection_items_path(c, status: "rejected_by_collection")
+  elsif item_status == "awaiting user approval"
+    visit collection_items_path(c, status: "awaiting_user_approval")
+  elsif item_status == "awaiting collection approval"
+    visit collection_items_path(c)
```
[the remark this answers is not in the document]
Comment (sarken) [reply features/step_definitions/collection_steps.rb:56]: Yeah, but I wanted to offer standard phrasing we could use for that page. I'll take out the `else` instead.

File: app/controllers/collection_items_controller.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (tickinginstant) [reply app/controllers/collection_items_controller.rb:23]: I made https://otwarchive.atlassian.net/browse/AO3-6507 for the refactor.

File: config/locales/helpers/en.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      user:
+        bookmark: Bookmarker approval status
+        work: Creator approval status
+  mailer_helper:
+      user_manage_collection_items_page_name:
+        approved: Manage Approved Collection Items page
+        rejected_by_collection: Manage Collection Items Rejected by Collection page
+        rejected_by_user: Manage Collection Items Rejected by User page
+        unreviewed_by_collection: Manage Collection Items Awaiting Collection Approval page
+        unreviewed_by_user: Manage Collection Items Awaiting User Approval page
```
Comment (tickinginstant) [inline config/locales/helpers/en.yml:28]: I think these might be a vestige of some older code — I don't see `mailer_helper` anywhere in this PR.

File: app/controllers/collection_items_controller.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (sarken) [reply app/controllers/collection_items_controller.rb:24]: Thanks!

File: features/step_definitions/collection_steps.rb [hunk 5]
```diff
@@ -42,15 +42,17 @@
   step %{I post the work}
 end
 
-When(/^I view the(?: ([^"]*)) collection items page for "(.*?)"$/) do |item_status, collection|
+When /^I view the(?: ([^"]*)) collection items page for "(.*?)"$/ do |item_status, collection|
```
Comment (tickinginstant) [inline features/step_definitions/collection_steps.rb:45]: If you're eliminating the default `else` option, the first param here should no longer be optional.

**Review by tickinginstant — APPROVED** [review]
One last quibble, but otherwise it looks good!

**Pull request merged**